### ISSUE TITLE
Uncaught (in promise) ReferenceError: process is not defined

### ISSUE THREAD

**Issue by draentropia** [minio/minio-js#1051]
Hi there!

When I go to call the Client constructor, I'm getting this error: `Uncaught (in promise) ReferenceError: process is not defined`

I was navigating to find the proper way to fix it but I didn't managed it. Did somebody faced something like this?

Here some info of my system:
Ubuntu 22.04
packages:
    "browserify-zlib": "^0.2.0",
    "buffer": "^6.0.3",
    "crypto-browserify": "^3.12.0",
    "http": "false",
    "minio": "^7.0.28",
    "path-browserify": "^1.0.1",
    "react": "^18.2.0",
    "react-app-rewired": "^2.2.1",
    "react-dom": "^18.2.0",
    "react-icons": "^4.4.0",
    "react-router-dom": "^6.3.0",
    "react-scripts": "^5.0.1",
    "stream-browserify": "^3.0.0",
    "stream-http": "^3.2.0",
    "styled-components": "^5.3.5",
    "timers": "^0.1.1",
    "url": "^0.11.0",
    "web-vitals": "^2.1.4"

Using react-app-rewired to fix some import error caused by minio:
module.exports = function override (config, env) {
    console.log('override')
    let loaders = config.resolve
    loaders.fallback = {
        "fs": false,
        "tls": false,
        "net": false,
        "http": require.resolve("stream-http"),
        "https": false,
        "zlib": require.resolve("browserify-zlib") ,
        "path": require.resolve("path-browserify"),
        "stream": require.resolve("stream-browserify"),
        //"util": false,
        "crypto": require.resolve("crypto-browserify"),
    }
    
    return config
}

**Comment by prakashsvmx** [comment]
looks like it is used in non node environment build. is it vite build?

**Comment by prakashsvmx** [comment]
Also a working react example with MinIO : https://github.com/prakashsvmx/minio-js-web-browser-example

**Comment by draentropia** [comment]
Is in ReactJS. To run it I use npm start. Should it work with npm build?

**Comment by draentropia** [comment]
as an autoresponse from my last comment. It didn't work.

**Comment by prakashsvmx** [comment]
please share minimal code/setup to replicate the issue. ?

**Comment by draentropia** [comment]
Found. It doesn't work with react-scripts version 5.

**Comment by prakashsvmx** [comment]
yeah. webpack 5 does not include node polyfills by default anymore. We will investigate this

**Comment by szzhou** [comment]
How to solve this problem

**Comment by StoicDeveloper** [comment]
@prakashsvmx Why was this closed, this still seems to be a problem.

**Comment by ShivamJoker** [comment]
Running into the same error using ViteJS.
